File: apps/cargo-tracker/backend/routingms/src/main/java/com/example/cargotracker/routing/domain/model/valueobjects/CargoType.java
package com.example.cargotracker.routing.domain.model.valueobjects;

/**
 * 航海が受け入れる貨物種別。
 *
 * <p><b>Booking の {@code CargoType} とは別の型</b>にする。同じ名前でも、Booking では
 * 「その貨物が何か」、Routing では「その航海が何を受け入れるか」で、値が増える理由も
 * 別になる。共有カーネルに列挙型は置かない（domain-model.md）。</p>
 */
public enum CargoType {
    /** 一般貨物。 */
    GENERAL,
    /** 危険物。 */
    HAZARDOUS,
    /** 冷凍・冷蔵貨物。 */
    REEFER
}

File: apps/cargo-tracker/frontend/src/shared/domain/cargoType.ts
/**
 * 貨物種別（domain-model.md「ユビキタス言語」）。
 *
 * <p><b>共有に置く。</b> 予約（S21）だけでなく見積（S12）も同じ種別を扱う。
 * 機能ごとに型を持つと、種別を足したときに片方だけが直る。</p>
 *
 * <p><b>Routing は冷凍を {@code REEFER} と呼ぶ。</b> 呼び名が違うので、
 * 経路設計の画面へ遷移するときは翻訳する（{@code RoutingWorklistPage}）。
 * バックエンドでは契約の受け側が翻訳する。</p>
 */
export type CargoType = 'GENERAL' | 'HAZARDOUS' | 'REFRIGERATED';
